### PR TITLE
Improve doc strings of all functions

### PR DESCRIPTION
**Pull request by lautarodapin**
#242 
I started to create the doc string, please let me know if I'm doing this corrently.

PD: i follow your suggestion, delete the fork and refork it.

### CODE REVIEW

File: snap7/common.py [hunk 1]
```diff
@@ -29,9 +29,11 @@ class ADict(dict):
...
 
 class Snap7Library:
-    """
-    Snap7 loader and encapsulator. We make this a singleton to make
-    sure the library is loaded only once.
+    """Snap7 loader and encapsulator. We make this a singleton to make
+        sure the library is loaded only once.
+
+    Attributes:
+        lib_location (str, Optional): full path to the `snap7.dll` file. Defaults to None
```
Comment (gijzelaerr) [inline snap7/common.py:36]: If I'm not mistaken, you don't need to include the types in the docstrings since they are taken from type hinting already.

**Review by gijzelaerr — CHANGES_REQUESTED** [review]

File: snap7/common.py (hunk 1 above)
Comment (lautarodapin) [reply snap7/common.py:36]: You are right.
I was following [this](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) example.
But according to [sphinx documentation](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#type-annotations) is not necessary. I will remove them.

File: snap7/common.py [hunk 2]
```diff
@@ -44,6 +46,14 @@ def __new__(cls, *args, **kwargs):
         return cls._instance
 
     def __init__(self, lib_location: Optional[str] = None):
+        """ Loads the snap7 library using ctypes cdll.
+
+        Args:
+            lib_location: full path to the `snap7.dll` file. Defaults to None
```
Comment (gijzelaerr) [inline snap7/common.py:52]: From the type information, one already knows the default value is none. I would add "optional".

File: snap7/common.py [hunk 3]
```diff
@@ -29,9 +29,11 @@ class ADict(dict):
...
 
 class Snap7Library:
-    """
-    Snap7 loader and encapsulator. We make this a singleton to make
-    sure the library is loaded only once.
+    """Snap7 loader and encapsulator. We make this a singleton to make
+        sure the library is loaded only once.
+
+    Attributes:
+        lib_location: full path to the `snap7.dll` file. Defaults to None
```
Comment (gijzelaerr) [inline snap7/common.py:36]: From the type information, one already knows the default value is none. I would add "optional".

**Review by gijzelaerr — CHANGES_REQUESTED** [review]

File: snap7/common.py (hunk 3 above)
Comment (lautarodapin) [reply snap7/common.py:36]: In case of optionals with a default value?

Comment (gijzelaerr) [reply snap7/common.py:36]: optional implies a default of `None`

**Comment by gijzelaerr** [comment]
looking awesome! thanks

**Pull request merged**